Library/Source/RuntimeImpl.cpp

```diff
@@ -237,15 +241,16 @@ namespace Babylon
             });
         };
 
-        Env env{ GetModulePath().u8string().data(), std::move(executeOnScriptThread) };
-
-        m_env = &env;
-        auto hostScopeGuard = gsl::finally([this] { m_env = nullptr; });
+        m_env = std::make_unique<Env>(GetModulePath().u8string().data(), std::move(executeOnScriptThread));
```

Comment:
So... If I'm understanding this correctly, is the motivation behind this change the desire to have the Env be created on the main thread (instead of the JavaScript thread) on Apple only? This seems kind of bizarre. Can you add some context?

Reply:
metal context init needs to happen on the main thread. So env must be created in the thread init.